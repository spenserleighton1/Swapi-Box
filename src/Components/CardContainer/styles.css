.card-container {
  width: 75vw;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  float: right;
  background-color: #19212A; }
