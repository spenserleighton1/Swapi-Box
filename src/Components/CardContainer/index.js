import React from 'react';
import Card from '../Card';
import PropTypes from 'prop-types';
import './styles.css';

const CardContainer = ({ favorite, cards }) => {
  const renderedCards = cards.map((card, index) => {
    return <Card 
      info={card}
      favorite={favorite}
      key={index} />;
  });

  return (
    <div className="card-container">
      { renderedCards }
    </div>
  );
};

CardContainer.propTypes = {
  cards: PropTypes.array.isRequired,
  favorite: PropTypes.func
};

export default CardContainer;