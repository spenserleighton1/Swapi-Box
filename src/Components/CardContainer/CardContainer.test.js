import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from './';

describe('CardContainer', () => {
  let wrapper;
  let mockPeopleData;

  beforeEach(() => {

    mockPeopleData = [{
      name: "Luke Skywalker",
      species: "Human",
      homeworld: "Tatooine",
      homeworldPopulation: "200000"
    }, {
      name: "Lisa Skywalker",
      species: "Human",
      homeworld: "Tatooine",
      homeworldPopulation: "200000"
    }];

    wrapper = shallow(<CardContainer 
      cards={ mockPeopleData } />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // it('should render different cards depending on whats passed to displayCards', () => {
  //   console.log('~~~~~~~~~~~~~~~~~~~~~~~', wrapper)
  //   wrapper.instance().displayCards('people')

  //   expect(wrapper.find('.people').length).toEqual(2)

  // });
});