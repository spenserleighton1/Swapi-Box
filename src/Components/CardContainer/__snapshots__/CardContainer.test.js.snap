// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    cards={
        Array [
            Object {
              "homeworld": "Tatooine",
              "homeworldPopulation": "200000",
              "name": "Luke Skywalker",
              "species": "Human",
            },
            Object {
              "homeworld": "Tatooine",
              "homeworldPopulation": "200000",
              "name": "Lisa Skywalker",
              "species": "Human",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          favorite={undefined}
          info={
                    Object {
                              "homeworld": "Tatooine",
                              "homeworldPopulation": "200000",
                              "name": "Luke Skywalker",
                              "species": "Human",
                            }
          }
/>,
        <Card
          favorite={undefined}
          info={
                    Object {
                              "homeworld": "Tatooine",
                              "homeworldPopulation": "200000",
                              "name": "Lisa Skywalker",
                              "species": "Human",
                            }
          }
/>,
      ],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "favorite": undefined,
          "info": Object {
            "homeworld": "Tatooine",
            "homeworldPopulation": "200000",
            "name": "Luke Skywalker",
            "species": "Human",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "favorite": undefined,
          "info": Object {
            "homeworld": "Tatooine",
            "homeworldPopulation": "200000",
            "name": "Lisa Skywalker",
            "species": "Human",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            favorite={undefined}
            info={
                        Object {
                                    "homeworld": "Tatooine",
                                    "homeworldPopulation": "200000",
                                    "name": "Luke Skywalker",
                                    "species": "Human",
                                  }
            }
/>,
          <Card
            favorite={undefined}
            info={
                        Object {
                                    "homeworld": "Tatooine",
                                    "homeworldPopulation": "200000",
                                    "name": "Lisa Skywalker",
                                    "species": "Human",
                                  }
            }
/>,
        ],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "favorite": undefined,
            "info": Object {
              "homeworld": "Tatooine",
              "homeworldPopulation": "200000",
              "name": "Luke Skywalker",
              "species": "Human",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "favorite": undefined,
            "info": Object {
              "homeworld": "Tatooine",
              "homeworldPopulation": "200000",
              "name": "Lisa Skywalker",
              "species": "Human",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
